core/iwasm/aot/aot_perf_map.c

```diff
@@ -31,8 +31,8 @@ get_func_size(const AOTModule *module, struct func_info *sorted_func_ptrs,
 static int
 compare_func_ptrs(const void *f1, const void *f2)
 {
-    return (intptr_t)((struct func_info *)f1)->ptr
-           - (intptr_t)((struct func_info *)f2)->ptr;
+    return (int)((intptr_t)((struct func_info *)f1)->ptr
+                 - (intptr_t)((struct func_info *)f2)->ptr);
```

Comment:
the original code looks broken. this change hides the issue.

Reply:
🤔 it is better?

```c
{
    void *ptr1 = ((struct func_info *)f1)->ptr;
    void *ptr2 = ((struct func_info *)f2)->ptr;

    if (ptr1 < ptr2)
        return -1;
    else if (ptr1 > ptr2)
        return 1;
    else
        return 0;
}
```